[DoctrineBridge] [DX] Improve condition for exception text in ManagerRegistry to avoid confusion

guilliamxavier opened the pull request:
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | yesish
| New feature?  | no
| Deprecations? | no
| Tickets       | Fix https://github.com/symfony/symfony/pull/31047#issuecomment-575548372
| License       | MIT
| Doc PR        | -

(Targetting 4.4 because 4.3 is already almost EOM)

guilliamxavier committed:
[DoctrineBridge] [DX] Improve condition for exception text in ManagerRegistry to avoid confusion

nicolas-grekas commented on src/Symfony/Bridge/Doctrine/ManagerRegistry.php:
```diff
@@ -50,7 +51,7 @@ protected function resetService($name)
 
         if (!$manager instanceof LazyLoadingInterface) {
-            throw new \LogicException('Resetting a non-lazy manager service is not supported. '.(interface_exists(LazyLoadingInterface::class) ? sprintf('Declare the "%s" service as lazy.', $name) : 'Try running "composer require symfony/proxy-manager-bridge".'));
+            throw new \LogicException('Resetting a non-lazy manager service is not supported. '.(interface_exists(LazyLoadingInterface::class) && class_exists(RuntimeInstantiator::class) ? sprintf('Declare the "%s" service as lazy.', $name) : 'Try running "composer require symfony/proxy-manager-bridge".'));
```
because LazyLoadingInterface is a required dependency of RuntimeInstantiator, I suppose we can check only the 2nd?

guilliamxavier replied:
That was my first idea (replace rather than augment), but on second thoughts I chose to play it safe and keep the existence check of the actually `instanceof`-tested interface (which also makes it clear that the code handles the possibility of it *not* existing).
But I have no strong opinion... However I won't be able to amend the PR (nor to rebase it to 4.3) soon (sorry I can only work via the web UI currently), but edits from maintainers are allowed ;)

nicolas-grekas reviewed (commented):
for 4.3 apparently

fabpot committed:
updated CHANGELOG for 3.4.37

fabpot committed:
update CONTRIBUTORS for 3.4.37

fabpot committed:
updated VERSION for 3.4.37

fabpot committed:
Merge pull request #35411 from fabpot/release-3.4.37

released v3.4.37

fabpot committed:
bumped Symfony version to 3.4.38

nicolas-grekas committed:
Fix testing with mongodb

nicolas-grekas committed:
minor #35450 Fix testing with mongodb (nicolas-grekas)

This PR was merged into the 3.4 branch.

Discussion
----------

Fix testing with mongodb

| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | -
| License       | MIT
| Doc PR        | -

Commits
-------

28cd964ac9 Fix testing with mongodb

nicolas-grekas committed:
Merge branch '3.4' into 4.3

* 3.4:
  Fix testing with mongodb
  bumped Symfony version to 3.4.38
  updated VERSION for 3.4.37
  update CONTRIBUTORS for 3.4.37
  updated CHANGELOG for 3.4.37

nicolas-grekas commented:
Thank you @guilliamxavier.

guilliamxavier commented:
Thank you and sorry for the extra work
